### PR TITLE
EDU-162 Change account page to my space

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-162

### CODE REVIEW

File: src/components/default-page-template.js [hunk 1]
```diff
@@ -65,59 +68,74 @@ function DefaultPageTemplate({ children, fullScreen, headerActions, alerts }) {
...
-  }
-
-  pageMenuItems.push({
-    key: 'language',
-    onClick: () => setIsUiLanguageDialogVisible(true),
-    text: t('common:language'),
-    icon: GlobalOutlined,
-    permission: null
-  });
+      permission: permissions.EDIT_SETTINGS,
```
Comment (laurentiu-ilici) [inline src/components/default-page-template.js:119]: This is unrelated to the PR itself: Why do we only show a help page to users that have this permission? This means only admins will ever see it, shouldn't we have help pages for everyone?

**Comment by ahelmberger** [comment]
While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).

File: src/components/default-page-template.js [hunk 2]
```diff
@@ -65,59 +68,74 @@ function DefaultPageTemplate({ children, fullScreen, headerActions, alerts }) {
       href: urls.getHomeUrl(),
       text: t('pageNames:home'),
       icon: HomeOutlined,
-      permission: null
+      permission: null,
+      showWhen: true
```
Comment (ahelmberger) [inline src/components/default-page-template.js:72]: Good idea!

**Comment by adrianaluchian** [comment]
> While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).

We could also make the profile tab as the first tab, as the simplest solution. 
Come to think of it, it probably does make sense to see this first.

Will check the insufficient profile warning case, see why it's not shown.

**Comment by adrianaluchian** [comment]
> > While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).
> 
> We could also make the profile tab as the first tab, as the simplest solution. Come to think of it, it probably does make sense to see this first.
> 
> Will check the insufficient profile warning case, see why it's not shown.

We do show the insufficient profile warning, to test please remove both first and last name from the profile and go to `docs` page for example. It is not shown on the `my-space` page.

File: src/components/profile-tab.yml [hunk 3]
```diff
@@ -1,49 +1,34 @@
...
-profile.country:
+country:
   en: Country
   de: Land
-profile.updateSuccessMessage:
+updateSuccessMessage:
   en: Your profile was updated successfully.
   de: Ihr Profil wurde erfolgreich aktualisiert.
-profile.avatarDescription:
+avatarDescription:
```
Comment (laurentiu-ilici) [inline src/components/profile-tab.yml:31]: This still says elmu, maybe we should make some ticket to make it overridable.

**Comment by ahelmberger** [comment]
> > > While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).
> > 
> > 
> > We could also make the profile tab as the first tab, as the simplest solution. Come to think of it, it probably does make sense to see this first.
> > Will check the insufficient profile warning case, see why it's not shown.
> 
> We do show the insufficient profile warning, to test please remove both first and last name from the profile and go to `docs` page for example. It is not shown on the `my-space` page.

True, thanks, I was also expecting it on the "fullScreen" pages like in the very old days :-) So yeah, I think the idea to make the profile the first tab would make a lot of sense.

**Comment by adrianaluchian** [comment]
> > > > While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).
> > > 
> > > 
> > > We could also make the profile tab as the first tab, as the simplest solution. Come to think of it, it probably does make sense to see this first.
> > > Will check the insufficient profile warning case, see why it's not shown.
> > 
> > 
> > We do show the insufficient profile warning, to test please remove both first and last name from the profile and go to `docs` page for example. It is not shown on the `my-space` page.
> 
> True, thanks, I was also expecting it on the "fullScreen" pages like in the very old days :-) So yeah, I think the idea to make the profile the first tab would make a lot of sense.

Done!

File: src/components/account-tab.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    forbiddenUsernames: []
+  });
+
+  const formRef = React.createRef();
+
+  const saveAccountData = async ({ username, email }) => {
+    try {
+      const { result, user: updatedUser } = await userApiClient.saveUserAccount({ username, email });
+
+      switch (result) {
```
Comment (laurentiu-ilici) [inline src/components/account-tab.js:37]: Maybe we should consider for such cases using the useReducer hook. It would make this testable and a lot easier to read. @adrianaluchian @ahelmberger what do you think?

**Comment by ahelmberger** [comment]
> > > > > While testing this I realized that in the educandu test app we do not show the insufficient profile warning (I wanted to test the link to what was the profile page before). It's not really related to this PR, but I think we should enable it. Something that might be a little bit of a downside is, that now when the user clicks on this link they get transferred to a page where the profile is shown in the second tab (and not immediately visible).
> > > > 
> > > > 
> > > > We could also make the profile tab as the first tab, as the simplest solution. Come to think of it, it probably does make sense to see this first.
> > > > Will check the insufficient profile warning case, see why it's not shown.
> > > 
> > > 
> > > We do show the insufficient profile warning, to test please remove both first and last name from the profile and go to `docs` page for example. It is not shown on the `my-space` page.
> > 
> > 
> > True, thanks, I was also expecting it on the "fullScreen" pages like in the very old days :-) So yeah, I think the idea to make the profile the first tab would make a lot of sense.
> 
> Done!

:tada:

**Review by ahelmberger — APPROVED** [review]

File: src/components/profile-tab.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+function ProfileTab({ formItemLayout, tailFormItemLayout }) {
+  const user = useUser();
+  const { language } = useLanguage();
+  const { t } = useTranslation('profileTab');
+  const userApiClient = useService(UserApiClient);
+  const countryNameProvider = useService(CountryNameProvider);
+
+  const profile = user.profile || { country: '' };
+  const gravatarUrl = gravatar.url(user.email, { s: AVATAR_SIZE, d: 'mp' });
+  const gravatarRagistrationUrl = `https://${language}.gravatar.com/`;
```
Comment (laurentiu-ilici) [inline src/components/profile-tab.js:40]: Maybe we could move this to the urls package? urls.getGravatarRegistrationUrl(language)?

Comment (adrianaluchian) [reply src/components/profile-tab.js:40]: To me that's for composing internal/website URLs, I would not put this one there.

File: src/components/default-page-template.js (hunk 1 above)
Comment (adrianaluchian) [reply src/components/default-page-template.js:119]: Yeah, interesting 🤔

File: src/components/profile-tab.yml (hunk 3 above)
Comment (adrianaluchian) [reply src/components/profile-tab.yml:31]: Also the browser tab title says ELMU, btw 😊 
The customization of educandu is not complete.

File: src/components/account-tab.js (hunk 4 above)
Comment (ahelmberger) [reply src/components/account-tab.js:37]: I don't think that it will be easier to read, wouldn't it be basically the same as we have here (just distributed into two places)?

File: src/components/default-page-template.js (hunk 1 above)
Comment (ahelmberger) [reply src/components/default-page-template.js:119]: That is a very good question! Nobody has noticed yet :laughing:

File: src/components/account-tab.js (hunk 4 above)
Comment (adrianaluchian) [reply src/components/account-tab.js:37]: Good to keep in mind 🤔 
In this case though I believe it would not make sense as it's a one-trigger event and represents the form save

File: src/components/profile-tab.yml (hunk 3 above)
Comment (ahelmberger) [reply src/components/profile-tab.yml:31]: Resources are already overridable

Comment (adrianaluchian) [reply src/components/profile-tab.yml:31]: But maybe we should make a config to pass in the project name 🤔 So that the implementing project does not have to overwrite each message.

Comment (laurentiu-ilici) [reply src/components/profile-tab.yml:31]: I guess that will be a another ticket though, I'll resolve this comment.

File: src/components/default-page-template.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/components/default-page-template.js:119]: I've made a ticket to reconsider, we can delete it later if we don't need it: https://educandu.atlassian.net/browse/EDU-203

**Review by laurentiu-ilici — APPROVED** [review]

File: src/components/default-page-template.js (hunk 1 above)
Comment (adrianaluchian) [reply src/components/default-page-template.js:119]: Good idea, let's reconsider then.
